app/src/components/SignInForm.js

```diff
@@ -0,0 +1,123 @@
+import React, {Component} from 'react'
+import isGuest from '../common/isGuest'
+import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
+import '../css/App.css';
+import {NavLink} from 'react-router-dom'
+import Alerts from './Alerts';
+import UsersDataRequests from "../common/UsersDataRequests";
+import endPointsList from '../common/endPointsList';
+
+class SignInForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            email: '',
+            password: '',
+            isLoggedIn: false
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange(event) {
+        this.setState({[event.target.name]: event.target.value});
+    }
+
+
+    async handleSubmit(event) {
+        event.preventDefault();
+        const user = {
+            email: this.state.email,
+            password: this.state.password
+        };
+
+        const validationResult = await this.validateForm(user);
+
+        try {
+            if (!validationResult) {
+                throw new Error(`Because we have some error we haven't any alerts`);
+            }
+        } catch (err) {
+            console.log(err);
+        }
+
+        this.setState({...this.state, alert: validationResult});
+        if (this.state.alert.type === 'success') {
+            this.setState({isLoggedIn: true});
+        }
+    }
+
+    async validateForm(user) {
+        if (!(user.email && user.password)) {
+            return {show: true, type: 'danger', tittle: 'Fields are empty!'}
+        } else {
+            const reqToDb = await UsersDataRequests.signInReq(user);
```

Comment:
validateForm method haven't to authorize user and do any requests.
It has to return only validation info.